File: 1301-1400/1371-Find the Longest Substring Containing Vowels in Even Counts/1371-Find the Longest Substring Containing Vowels in Even Counts.cpp
class Solution {
public:
    int findTheLongestSubstring(string s) {
        unordered_map<int, int> table{{0, -1}};
        int maxLen = 0, curr = 0;
        string candidate = "aeiou";
        for (int i = 0; i < s.size(); ++i) {
            int index = candidate.find(s[i]);
            if (index != string::npos) {
                curr ^= 1 << index;
            }
            if (!table.count(curr)) {
                table[curr] = i;
            }
            else {
                maxLen = max(maxLen, i - table[curr]);
            }            
        }
        return maxLen;
    }
};

File: 1301-1400/1371-Find the Longest Substring Containing Vowels in Even Counts/1371-Find the Longest Substring Containing Vowels in Even Counts.py
class Solution:
    def findTheLongestSubstring(self, s: str) -> int:
        table = {0 : -1}
        curr = maxLen = 0
        for i, c in enumerate(s):
            index = 'aeiou'.find(c)
            if index != -1:
                curr ^= 1 << index
            if curr not in table:
                table[curr] = i
            else:
                maxLen = max(maxLen, i - table[curr])
        return maxLen
